Cannot build 0.1.105 on FreeBSD

squashfuse 0.1.104 is in FreeBSD Ports. But I need 0.1.105 so that I can use `ll.h`. So I edited the Makefile a bit:

```
PORTNAME=       squashfuse
PORTVERSION=    0.1.105
CATEGORIES=     sysutils
MASTER_SITES=   ${MASTER_SITE_GITHUB}/vasi/${PORTNAME}/archive/refs/tags/
DISTNAME=       ${PORTVERSION}
EXTRACT_SUFX=   .tar.gz
PKGNAMEPREFIX=  fusefs-

MAINTAINER=     jsmith@resonatingmedia.com
COMMENT=        Mount a squashfs archive and treat it like a local file system
WWW=            https://github.com/vasi/squashfuse

LICENSE=        BSD2CLAUSE
LICENSE_FILE=   ${WRKSRC}/LICENSE

# Specify the direct URL to the tarball
DISTFILES=      ${DISTNAME}${EXTRACT_SUFX}

USES=           autoreconf fuse libtool pkgconfig
GNU_CONFIGURE=  yes

INSTALL_TARGET= install-strip

# Set the correct path to the extracted source code directory
WRKSRC=         ${WRKDIR}/${PORTNAME}-${PORTVERSION}

.include <bsd.port.mk>
```

But the build fails with:

```
--- squashfuse_ll-ll_main.o ---
In file included from ll_main.c:25:
--- squashfuse-hl.o ---
In file included from hl.c:26:
./fuseprivate.h:31:10: fatal error: 'fuse3/fuse.h' file not found
#include <fuse3/fuse.h>
         ^~~~~~~~~~~~~~
--- squashfuse_ll-ll_main.o ---
./ll.h:31:10: fatal error: 'fuse3/fuse_lowlevel.h' file not found
#include <fuse3/fuse_lowlevel.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
--- squashfuse_extract-nonstd-makedev.o ---
```

There is no fuse3 in FreeBSD:

```
# pkg list fusefs-libs
/usr/local/include/fuse.h
/usr/local/include/fuse/cuse_lowlevel.h
/usr/local/include/fuse/fuse.h
/usr/local/include/fuse/fuse_common.h
/usr/local/include/fuse/fuse_common_compat.h
/usr/local/include/fuse/fuse_compat.h
/usr/local/include/fuse/fuse_kernel.h
/usr/local/include/fuse/fuse_lowlevel.h
/usr/local/include/fuse/fuse_lowlevel_compat.h
/usr/local/include/fuse/fuse_opt.h
/usr/local/include/ulockmgr.h
/usr/local/lib/libfuse.a
/usr/local/lib/libfuse.so
/usr/local/lib/libfuse.so.2
/usr/local/lib/libfuse.so.2.9.9
/usr/local/lib/libulockmgr.a
/usr/local/lib/libulockmgr.so
/usr/local/lib/libulockmgr.so.1
/usr/local/lib/libulockmgr.so.1.0.1
/usr/local/libdata/pkgconfig/fuse.pc
/usr/local/share/doc/fusefs/libs/how-fuse-works
/usr/local/share/doc/fusefs/libs/kernel.txt
/usr/local/share/licenses/fusefs-libs-2.9.9_2/LGPL21
/usr/local/share/licenses/fusefs-libs-2.9.9_2/LICENSE
/usr/local/share/licenses/fusefs-libs-2.9.9_2/catalog.mk
```

Is it true that from 0.1.104 to 0.1.105 libfuse3 became a requirement? This would be a bit unexpected for a patch level release.

cc @Jehops

--- a/fuseprivate.c
+++ b/fuseprivate.c
@@ -32,6 +32,10 @@
 
 #include "nonstd.h"
 
+#if HAVE_DECL_FUSE_CMDLINE_HELP
+    #include <fuse_lowlevel.h>
+#endif
+
 int sqfs_listxattr(sqfs *fs, sqfs_inode *inode, char *buf, size_t *size) {
 	sqfs_xattr x;
 	size_t count = 0;
@@ -83,7 +87,7 @@ void sqfs_usage(char *progname, bool fuse_usage, bool ll_usage) {
 	if (fuse_usage) {
 		if (ll_usage) {
 #ifdef SQFS_MULTITHREADED
-#if FUSE_USE_VERSION >= 30
+#if HAVE_DECL_FUSE_CMDLINE_HELP
 			fprintf(stderr, "\nFUSE options:\n");
 			fuse_cmdline_help();
 #else

--- a/fuseprivate.h
+++ b/fuseprivate.h
@@ -27,12 +27,7 @@
 
 #include "squashfuse.h"
 
-#if FUSE_USE_VERSION >= 30
-#include <fuse3/fuse.h>
-#include <fuse3/fuse_lowlevel.h>
-#else
 #include <fuse.h>
-#endif
 
 /* Common functions for FUSE high- and low-level clients */
 

--- a/ll.h
+++ b/ll.h
@@ -27,11 +27,7 @@
 
 #include "squashfuse.h"
 
-#if FUSE_USE_VERSION >= 30
-#include <fuse3/fuse_lowlevel.h>
-#else
 #include <fuse_lowlevel.h>
-#endif
 
 typedef struct sqfs_ll sqfs_ll;
 struct sqfs_ll {

--- a/m4/squashfuse_fuse.m4
+++ b/m4/squashfuse_fuse.m4
@@ -271,6 +271,9 @@ AC_DEFUN([SQ_FUSE_API_VERSION],[
 			AC_DEFINE([HAVE_NEW_FUSE_UNMOUNT],1,
 					[Define if we have two-argument fuse_unmount])
 		])
+
+		AC_CHECK_DECLS([fuse_cmdline_help],,,
+		        [#include <fuse_lowlevel.h>])
 	])
 	
 	SQ_RESTORE_FLAGS

--- a/nonstd-daemon.c
+++ b/nonstd-daemon.c
@@ -28,11 +28,7 @@
 #include "nonstd-internal.h"
 
 #include <unistd.h>
-#if FUSE_USE_VERSION >= 30
-#include <fuse3/fuse_lowlevel.h>
-#else
 #include <fuse_lowlevel.h>
-#endif
 
 int sqfs_ll_daemonize(int fg) {
 	#if HAVE_DECL_FUSE_DAEMONIZE

